### PR TITLE
Json format on doc_search, doc_summary, and google_gpt

### PR DESCRIPTION
**Pull request by SanderGi**
Task: https://app.asana.com/0/1203067047205953/1207720614246546
Only for gpt4o, gpt4 turbo vision, and gpt4 turbo

### Q/A checklist

- [x] If you add new dependencies, did you update the lock file?
```bash
poetry lock --no-update
```
- [x] Run tests 
```bash
ulimit -n unlimited && ./scripts/run-tests.sh
```
- [x] Do a self code review of the changes - Read the diff at least twice.  
- [x] Carefully think about the stuff that might break because of this change - this sounds obvious but it's easy to forget to do "Go to references" on each function you're changing and see if it's used in a way you didn't expect. 
- [x] The relevant pages still run when you press submit
- [x] The API for those pages still work (API tab)
- [x] The public API interface doesn't change if you didn't want it to (check API tab > docs page)
- [x] Do your UI changes (if applicable) look acceptable on mobile?
- [x] Ensure you have not regressed the import time unless you have a good reason to do so. 
You can visualize this using tuna:
```bash
python3 -X importtime -c 'import server' 2> out.log && tuna out.log
```
To measure import time for a specific library:
```bash
$ time python -c 'import pandas'

________________________________________________________
Executed in    1.15 secs    fish           external
   usr time    2.22 secs   86.00 micros    2.22 secs
   sys time    0.72 secs  613.00 micros    0.72 secs
```
To reduce import times, import libraries that take a long time inside the functions that use them instead of at the top of the file:
```python
def my_function():
    import pandas as pd
    ...
```

### Legal Boilerplate

Look, I get it. The entity doing business as “Gooey.AI” and/or “Dara.network” was incorporated in the State of Delaware in 2020 as Dara Network Inc. and is gonna need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Dara Network Inc can use, modify, copy, and redistribute my contributions, under its choice of terms.

### CODE REVIEW

File: daras_ai_v2/language_model.py [hunk 1]
```diff
@@ -143,6 +147,7 @@ class LargeLanguageModels(Enum):
         llm_api=LLMApis.openai,
         context_window=4096,
         price=1,
+        supports_json=True,
```
Comment (SanderGi) [inline daras_ai_v2/language_model.py:151]: Have you tested this? When I went through and tested all the open ai models, this one did not work for me 🤔

File: daras_ai_v2/language_model.py [hunk 2]
```diff
@@ -208,20 +213,23 @@ class LargeLanguageModels(Enum):
...
     )
     gemma_7b_it = LLMSpec(
         label="Gemma 7B (Google)",
         model_id="gemma-7b-it",
         llm_api=LLMApis.groq,
         context_window=8_192,
         price=1,
+        supports_json=True,
```
Comment (SanderGi) [inline daras_ai_v2/language_model.py:232]: Shouldn’t this include validation and retries to make sure the output is JSON similar to the openai implementation?

Comment (devxpy) [reply daras_ai_v2/language_model.py:232]: groq supports json mode

Comment (devxpy) [reply daras_ai_v2/language_model.py:232]: https://console.groq.com/docs/text-chat#json-mode-object-object

File: daras_ai_v2/language_model.py (hunk 1 above)
Comment (devxpy) [reply daras_ai_v2/language_model.py:151]: I just copied the ones listed here - https://platform.openai.com/docs/guides/json-mode

Comment (devxpy) [reply daras_ai_v2/language_model.py:151]: in any case it should work for all models to some degree because of the added json prompt

Comment (SanderGi) [reply daras_ai_v2/language_model.py:151]: Yes, but when I run gpt 3.5 with the json flag enabled, it errors out in practice: https://gooey.ai/compare-large-language-models/?run_id=oygoftsplu02&uid=2pZJQouaP2Okm18cbEhIpOQpqCg1

Comment (devxpy) [reply daras_ai_v2/language_model.py:151]: i guess not though, it might lead to a misuse of json mode on non json mode models

Comment (SanderGi) [reply daras_ai_v2/language_model.py:151]: The supports_json flag above was meant to be used to disable json mode for models that don't support it. I'll leave it up to you whether we prefer showing the JSON format ui but throwing an error in that case (or perhaps even implement our own JSON mode that retries a couple of times if the response happens to be invalid json) over ignoring it for unsupported models

Comment (devxpy) [reply daras_ai_v2/language_model.py:151]: What we have now is ok. 

PS doing json mode well requires constrained decoding https://lmsys.org/blog/2024-02-05-compressed-fsm/

**Pull request merged**